### PR TITLE
First draft for the Boston Light Swim post

### PR DESCRIPTION
**Pull request by ocefpaf**
@rsignell-usgs this is just a compilation of the 3 notebooks into one. I will make some adjustments to make it more appealing for publication in the site. Also, I will not remove the 3-part version from staged notebooks.

### CODE REVIEW

**Comment by ocefpaf** [comment]
@rsignell-usgs and @mwengren please take a look at this notebook, particularly at `cell 5`:

```
********************* Catalog information **********************
URL: https://www.ngdc.noaa.gov/geoportal/csw
Number of datasets available: 1
HYbrid Coordinate Ocean Model (HYCOM): Global


URL: https://data.ioos.us/csw
Number of datasets available: 10
NAM CONUS 20km/Best NAM CONUS 20km Time Series
Rapid Refresh CONUS 13km/Best Rapid Refresh CONUS 13km Time Series
DGEX CONUS 12km/Best DGEX CONUS 12km Time Series
NAM CONUS 80km/Best NAM CONUS 80km Time Series
NOAA/NCEP Global Forecast System (GFS) Atmospheric Model
NAM Alaska 11km/Best NAM Alaska 11km Time Series
FNMOC COAMPS Western Atlantic/Best FNMOC COAMPS Western Atlantic Time Series
NAM Alaska 45km from NOAAPORT/Best NAM Alaska 45km from NOAAPORT Time Series
NAM Alaska 22km/Best NAM Alaska 22km Time Series
Rapid Refresh CONUS 40km/Best Rapid Refresh CONUS 40km Time Series
***************************** DAP ******************************
http://oos.soest.hawaii.edu/thredds/dodsC/hioos/model/atm/ncep_global/NCEP_Global_Atmospheric_Model_best.ncd.html
http://thredds.ucar.edu/thredds/dodsC/grib/FNMOC/COAMPS/Western_Atlantic/Best.html
http://thredds.ucar.edu/thredds/dodsC/grib/NCEP/DGEX/CONUS_12km/Best.html
http://thredds.ucar.edu/thredds/dodsC/grib/NCEP/NAM/Alaska_11km/Best.html
http://thredds.ucar.edu/thredds/dodsC/grib/NCEP/NAM/Alaska_22km/Best.html
http://thredds.ucar.edu/thredds/dodsC/grib/NCEP/NAM/Alaska_45km/noaaport/Best.html
http://thredds.ucar.edu/thredds/dodsC/grib/NCEP/NAM/CONUS_20km/noaaport/Best.html
http://thredds.ucar.edu/thredds/dodsC/grib/NCEP/NAM/CONUS_80km/Best.html
http://thredds.ucar.edu/thredds/dodsC/grib/NCEP/RAP/CONUS_13km/Best.html
http://thredds.ucar.edu/thredds/dodsC/grib/NCEP/RAP/CONUS_40km/Best.html


URL: http://gamone.whoi.edu/csw
Number of datasets available: 0
```

The `http://gamone.whoi.edu/csw` is not adding anything so I guess we can remove it. The global HYCOM model only shows up in the NGDC catalog, which we should also remove BTW.

The IOOS catalog is rturning some odd endpoints that I am not sure should be valid for the search in question: sea surface temperatura in the Boston Harbor.

Note also that two models we expected, `SMAST FVCOM` and `COASWST`, are not found anywhere.

**Comment by rsignell-usgs** [comment]
I removed my scanning of the PACIOOS thredds server and the NECOFS thredds server because I thought they were already in the IOOS catalog:
https://github.com/USGS-CMG/usgs-cmg-portal/commit/8cf3063faaf9cbdd615218d2261852e67fcd252c

Should I put them back into gamone's scan?

The reason we didn't find the COAWST data was unrelated: our FRMC aggregation stopped updating on our TDS, and thus we didn't find anything within the last 5 days (see https://github.com/Unidata/thredds/issues/719). 
It is there now.

**Comment by rsignell-usgs** [comment]
Okay, the problem with the NECOFS data is it's not updating daily.  The stop date in the metadata is from Dec 2!
https://data.ioos.us/dataset/necofs-massachusetts-fvcom-massachusetts-coastal-latest-forecast
Go to the ISO metadata record:
https://data.ioos.us/harvest/object/61380292-c76d-45eb-8b8b-3314acde1654
and search for `<gml:endPosition>`:
```
<gml:endPosition>2016-12-02T00:00:00Z</gml:endPosition>
```
@mwengren , @robragsdale, or @ebridger,  can you fix?

**Comment by rsignell-usgs** [comment]
The HYCOM model *is* in the IOOS catalog
https://data.ioos.us/dataset/hybrid-coordinate-ocean-model-hycom-global
 and the start/end dates found in the metadata:
```
https://data.ioos.us/harvest/object/da89da1c-f17c-4cb0-b5cc-6ec6740aeff6
```
should enable discovery:
```
<gmd:temporalElement>
<gmd:EX_TemporalExtent>
<gmd:extent>
<gml:TimePeriod gml:id="d81e100">
<gml:beginPosition>2008-05-08T00:00:00Z</gml:beginPosition>
<gml:endPosition>2017-12-31T00:00:00Z</gml:endPosition>
</gml:TimePeriod>
</gmd:extent>
</gmd:EX_TemporalExtent>
</gmd:temporalElement>
```
So I wonder why we are not picking it up?

**Comment by ebridger** [comment]
@rsignell-usgs, @mwengren Just checking the NECOFS WAF creation now, and the log file I create when running ncISO.jar on the NECOFS models is showing:
```
Error on line 1 column 50 of UnidataDDCount-HTML.xsl:
  SXXP0003: Error reported by XML parser: White spaces are required between publicId and systemId.
Error on line 1 column 50 of UnidataDD2MI.xsl:
  SXXP0003: Error reported by XML parser: White spaces are required between publicId and systemId.
Error on line 1 column 50 of UnidataDD2MI.xsl:
  SXXP0003: Error reported by XML parser: White spaces are required between publicId and systemId.
Error on line 1 column 50 of UnidataDD2MI.xsl:
  SXXP0003: Error reported by XML parser: White spaces are required between publicId and systemId.
Error on line 1 column 50 of UnidataDD2MI.xsl:
  SXXP0003: Error reported by XML parser: White spaces are required between publicId and systemId.
tds_url: http://www.smast.umassd.edu:8080/thredds/forecasts.xml
waf_dir: /home2/git2/odp1drupal7/htdocs/WAF/UMASSD-SMAST/Forecasts

```
Seems like I've seen messages from Rich with similar complaints on other TDS ncISO.jar WAF creations. The ISO xml files created are all 0 bytes.

**Comment by ebridger** [comment]
@rsignell-usgs @mwengren @robragsdale I've fixed the [NERACOOS WAF ](http://www.neracoos.org/WAF/) to use the thredds_crawler and not ncISO.jar The NECOFS  Forecast WAF is in UMASSD-SMAST/Forecasts/iso and gets updated twice daily.  I don't think I've changed the directory structure so hopefully the registry will pick up the new files.

**Comment by ocefpaf** [comment]
@jbosch-noaa this is ready for review!

hypothes.is link: 
https://via.hypothes.is/http://nbviewer.jupyter.org/github/ocefpaf/notebooks_demos/blob/boston_light_swim/notebooks/2016-12-22-boston_light_swim.ipynb

**Comment by rsignell-usgs** [comment]
@jbosch-noaa , just to let you know, this PR fixes my issues on Windows.  Woohoo!

**Comment by rsignell-usgs** [comment]
@jbosch-noaa, okay to merge this and do any slight mods to text later?

**Comment by ocefpaf** [comment]
@rsignell-usgs I am working on @jbosch-noaa review now. Can this wait a few hours? Maybe less...

**Comment by ocefpaf** [comment]
> @rsignell-usgs I am working on @jbosch-noaa review now. Can this wait a few hours? Maybe less...

Done in https://github.com/ioos/notebooks_demos/pull/130/commits/449fe19020fa2e22d95f12cbadc7ec181bcefd10 @jbosch-noaa and @rsignell-usgs back to you :wink:

File: notebooks/2016-12-22-boston_light_swim.ipynb [hunk 1]
```diff
@@ -1298,8 +1286,7 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "Here we use a dictionary with some model we expect to find to create a better legend for the plots.\n",
-    "If any new model is found we will use its filename as legend instead."
+    "Here we use a dictionary with some models we expect to find so we can create a better legend for the plots. If any new model is found, we will use its filename in the legend as a default until we can go back and add a short name to our library."
```
Comment (jbosch-noaa) [inline notebooks/2016-12-22-boston_light_swim.ipynb]: line 1289
 "If any new models are found,"

File: notebooks/2016-12-22-boston_light_swim.ipynb [hunk 2]
```diff
@@ -283,9 +283,12 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "We only want to find model data using the catalog.\n",
-    "We'll rely `CO-OPS` and `NDBC` services for the observations.\n",
-    "This extra filtering step is performed by `is_station` below."
+    "We found some models, and observations from NERACOOS there.\n",
+    "However, we do know that there are some buoys from NDBC and CO-OPS available too.\n",
+    "Also, those NERACOOS observations seems to be from a [CTD](http://www.neracoos.org/thredds/dodsC/UMO/DSG/SOS/A01/CTD1m/HistoricRealtime/Agg.ncml.html) mounted at 65 meters below the sea surface. Rendering them useless from our purpose.\n",
```
Comment (jbosch-noaa) [inline notebooks/2016-12-22-boston_light_swim.ipynb]: line 288
"Also, those NERACOOS observations seem to be ..."

File: notebooks/2016-12-22-boston_light_swim.ipynb (hunk 1 above)
Comment (jbosch-noaa) [inline notebooks/2016-12-22-boston_light_swim.ipynb]: Just a few minor typos :)

**Comment by ocefpaf** [comment]
Should be good to go now. Thanks @jbosch-noaa for the review!

**Comment by rsignell-usgs** [comment]
Woohoo!!  Thanks for bringing this horse back to the stable!

**Pull request merged**